Integrations/IronDefense/IronDefense.py

```diff
@@ -241,14 +403,234 @@ def report_observed_bad_activity(self, name, description='', ip='', domain='',
+        }
+        response = self._http_request('POST', '/GetAlertIronDomeInformation', body=json.dumps(req_body))
+        if response.status_code != 200:
+            err_msg = self._get_error_msg_from_response(response)
+            self.logger.error('Failed to retrieve IronDome information for alert with ID ({}). The response failed '
+                              'with status code {}. The response was: {}'.format(alert_id, response.status_code,
+                                                                                 response.text))
+            raise HTTPException('Failed to retrieve IronDome information for alert with ID {} ({}): {}'
+                                .format(alert_id, response.status_code, err_msg))
+        else:
+            self.logger.debug('Successfully retrieved IronDome information for alert ({})'.format(alert_id))
+            dome_alert_info = response.json()
+            return dome_alert_info
+
 
 ''' COMMANDS MANAGER / SWITCH PANEL '''
 
 
-# This will be enabled at a later date.
-# def fetch_incidents_command():
-#     incidents = IRON_DEFENSE.fetch_incidents()
-#     demisto.incidents(incidents)
+def fetch_incidents_command():
+    # IronDome Notification related params
+    dome_categories = PARAMS.get('domeCategories', None)
+    dome_limit = int(PARAMS.get('domeLimit', 500))
+    disable_dome_notifs = not PARAMS.get('enableDomeNotifications', False)
+    # Alert Notification related params
+    alert_categories = PARAMS.get('alertCategories', None)
+    alert_subcategories = PARAMS.get('alertSubCategories', None)
+    alert_severity_lower = PARAMS.get('alertSeverityLower', None)
+    alert_severity_upper = PARAMS.get('alertSeverityUpper', None)
+    alert_limit = int(PARAMS.get('alertLimit', 500))
+    alert_actions = PARAMS.get('alertActions', None)
+    disable_alert_notifs = not PARAMS.get('enableAlertNotifications', True)
+    # Event Notification related params
+    event_categories = PARAMS.get('eventCategories', None)
+    event_subcategories = PARAMS.get('eventSubCategories', None)
+    event_severity_lower = PARAMS.get('eventSeverityLower', None)
+    event_severity_upper = PARAMS.get('eventSeverityUpper', None)
+    event_limit = int(PARAMS.get('eventLimit', 500))
+    event_actions = PARAMS.get('eventActions', None)
+    disable_event_notifs = not PARAMS.get('enableEventNotifications', False)
+
+    incidents: list = []
+    if disable_dome_notifs and disable_alert_notifs and disable_event_notifs:
+        LOGGER.debug("Ingestion of all notifications (Dome, Alert, Event) is disabled, not fetching")
+    else:
+        if disable_dome_notifs:
+            LOGGER.debug('Ingestion of Dome Notifications is disabled')
+        else:
+            incs = IRON_DEFENSE.fetch_dome_incidents(dome_categories, dome_limit)
+            incidents.extend(incs)
+            # If the limit was reached, poll again
+            poll_count = 1
+            while len(incs) == dome_limit and poll_count < 10:
```

Comment:
```suggestion
            while len(incs) <= dome_limit and poll_count < 10:
```

Reply:
I don't agree with this change. We only want to poll again if the previous call reached the limit. If the last call came in under the limit we assume we're at the end of the queue.